doc/configuration/Configuration.md

```diff
@@ -0,0 +1,319 @@
+
+
+### File Settings
+___
+Settings to configure storage, appearance, and security of files and images.
+
+### File Storage
+
+```"DriverName": "local"```
+*System used for file storage. “local”: Files and images are stored on the local file system. “amazons3”: Files and images are stored on Amazon S3 based on the provided access key, bucket and region fields.*
+
+```"Directory": "./data/"```
+*Directory to which files are written. If blank, directory will be set to ./data/.*
+
+```"AmazonS3AccessKeyId": ""```
+*Obtain this credential from your Amazon EC2 administrator.*
+
+```"AmazonS3SecretAccessKey": ""```
+*Obtain this credential from your Amazon EC2 administrator.*
+
+```"AmazonS3Bucket": ""```
+*Name you selected for your S3 bucket in AWS.*
+
+```"AmazonS3Region": ""```
+*AWS region you selected for creating your S3 bucket.*
+
+### Image Settings
+
+```"ThumbnailWidth": 120```
+*Width of thumbnails generated from uploaded images. Updating this value changes how thumbnail images render in future, but does not change images created in the past.*
+
+```"ThumbnailHeight": 100```
+*Height of thumbnails generated from uploaded images. Updating this value changes how thumbnail images render in future, but does not change images created in the past.*
+
+```"PreviewWidth": 1024```
+*Maximum width of preview image. Updating this value changes how preview images render in future, but does not change images created in the past.*
+
+```"PreviewHeight": 0```
+*Maximum height of preview image ("0": Sets to auto-size). Updating this value changes how preview images render in future, but does not change images created in the past.*
+
+```"ProfileWidth": 128```
+*Width of a profile picture recommended when uploading from Account Settings. A users full size profile picture is shown when you hover over their name in a channel thread.*
+
+```"ProfileHeight": 128```
+*Height of a profile picture recommended when uploading from Account Settings. A users full size profile picture is shown when you hover over their name in a channel thread.*
+
+```"EnablePublicLink": true```
+*”true”: Allow users to share public links to files and images when previewing. “false”*
+
+```"PublicLinkSalt": "A705AklYF8MFDOfcwh3I488G8vtLlVip"```
+*32-character (to be randomly generated via Admin Console) salt added to signing of public image links.*
+
+
+### Log Settings
+___
+Settings to configure the console and log file output, detail level, format and location of error messages.
+
+### Console Settings
+
+```"EnableConsole": true```
+*“true”: Output log messages to the console based on ConsoleLevel option. The server writes messages to the standard output stream (stdout).*
+
+```"ConsoleLevel": "DEBUG"```
+*Level of detail at which log events are written to the console when EnableConsole=true. ”ERROR”: Outputs only error messages; “INFO”: Outputs error messages and information around startup and initialization; “DEBUG”: Prints high detail for developers debugging issues.*
```

Comment:
Might be good to add formatting around setting names in the description. Maybe italics or code blocking, not sure